### PR TITLE
Improve observability in Sentry

### PR DESCRIPTION
**Pull request by jonfreeland**
### Why

This PR aims to improve observability. 

It adds release and deployment tracking to Sentry, as well as sourcemaps.

It creates a build script for Render to use, instead of managing the build command in each individual Render site. This build script will pass the current release (commit SHA) as an environment variable on build, so it's reported properly to Sentry at runtime.

Note: Render sites will need to be updated to use this script after this PR is merged.

### Ticket

See below.

### Accomplishments

- [x] Created a build script for Render to execute
- [x] Track releases and deployments in Sentry
- [x] Upload sourcemaps to Sentry
- [x] Send additional user context to Sentry

### Environment variables

Added an environment variable needed at build _only_: `REACT_APP_VERSION`.

### CODE REVIEW

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -49,3 +54,31 @@ jobs:
           cache: "yarn"
       - run: yarn install
       - run: CI=false yarn run build
+
+      - name: Notify Sentry release is available
```
Comment (Ronan-ACN) [inline .github/workflows/cicd.yml:58]: should we switch to https://github.com/marketplace/actions/sentry-release ?

Comment (jonfreeland) [reply .github/workflows/cicd.yml]: Moved some things around.

Comment (jonfreeland) [reply .github/workflows/cicd.yml]: If this works, will do the same for other repos.

**Review by Ronan-ACN — APPROVED** [review]

File: src/AppAuthenticator.tsx [hunk 2]
```diff
@@ -482,6 +482,8 @@ const AppAuthenticator: React.FC = ({ children }) => {
 
           Sentry.setUser({
             id: payload.userId,
+            email: payload.email,
```
Comment (lvancraen) [inline src/AppAuthenticator.tsx:485]: Do we want to do an equivalent for passwordless users?

Comment (jonfreeland) [reply src/AppAuthenticator.tsx:485]: Good point... Sentry doesn't have another first-class property that would work for phone number. I'll look around a bit more.

**Review by stephanie-saretsky — APPROVED** [review]

**Pull request merged**